run_unittests.py

```diff
@@ -5627,6 +5627,22 @@ def test_clang_format(self):
         self.assertEqual(0, output.count('File reformatted:'))
         self.build('clang-format-check')
 
+    def test_compiler_args_added_to_linker_args(self) -> None:
+        # If the linker is invoked by the compiler then we need to add compiler
+        # arguments to the linker arguments, otherwise we should not. This is
```

Comment:
I still do not agree with this and it is not how things were originally implemented and documented which is:

- c_args is used for compilation only
- c_link_args is used for linking only
- when parsing system envvars, CPPFLAGS is put only to c_args, but CFLAGS is put to both c_args and c_link_args

This is compatible with the "classic" disto envvar spamming approach but still keeps a strict separation between the two internally. It should also be noted that there are a many compile args that make no sense to add to the linker command, like -DFOO or -O2.

Reply:
Sure, but for gcc if i have, for example: `-DFOO -flto` in my c_args, passing `-DFOO` to gcc as a linker argument doesn't hurt anything, and not passing `-flto` breaks everything.

And having `$CFLAGS` and `-Dc_args=` have different behavior (which I cannot find documented anywhere) is bad UI, it's the sort of subtlety that an end user is likely to be unaware of until things stop working, and they can't figure out why.

Reply:
It should be noted that `CPPFLAGS` should only be used during compilation. If we were to always pass `c_args`, then this would be impossible to support.